src/relax/op/tensor/binary.cc

```diff
@@ -58,7 +56,13 @@ namespace relax {
 template <typename FType>
 StructInfo InferStructInfoBroadcast(const Call& call, const BlockBuilder& ctx,
                                     FType f_compute_out_dtype) {
-  auto [lhs_sinfo, rhs_sinfo] = GetBinaryInputTensorStructInfo(call, ctx, /*op_name=*/"Binary");
+  Array<TensorStructInfo> input_sinfo = GetInputTensorStructInfo<std::function<const char*(int)>>(
+      call, ctx, /*n_input=*/2, [](int i) {
+        ICHECK(i == 0 || i == 1);
```

Comment:
we can use https://github.com/tlc-pack/relax/blob/relax/include/tvm/ir/op.h#L70 which contains all the information needed